### ISSUE TITLE
1.2.0 release

### ISSUE THREAD

**Issue by BobDickinson** [pkgcloud/pkgcloud#494]
Please release 1.2.0 (publish to npm)

TL;DR

I'm new to this project (just started using it a week ago).  It solves a huge problem for us.  We'd love to see it thrive and find ways to help.

I spent some time trying to assess the state of the project, and frankly, I'm not sure I have any better understanding than when I started.  I reviewed all the open issues and open PRs, all the issues closed and PRs merged since the last release, and all of the changelists since the last release. 

It seems like there is still a core group of maintainers who are active.  It also seems like the changes are generally maintenance/bug fixes, or adding support for new service functionality within the scope of how and what pkgcloud already provided in terms of API (not a lot of big breaking changes, AFAICT).  

The last release published to npm was November of 2014.  Not only was that release a lifetime ago, but it was labelled "alpha.0".  That release came at the end of what was, until that time, a very nice release cadence (about once a month over the prior year).

So if we have active maintainers and a good set of CI tests, and are essentially in maintenance mode, why no release in 15+ months?  A lot of people are submitting PRs, presumably because they use pkgcloud and need stuff fixed, and those PRs don't exactly get jumped on, and even if they do get merged, they never hit npm.  I'm not sure how much risk there is of new bugs making it into a new npm release, but I can tell you that all of the fixed bugs represent a pretty good argument for a new release.

It's also concerning that the CI tests have been failing for 3 months due to a relatively simple issue (the version of blanket, AFAICT), and nobody has dealt with that issue (many recent PRs that fail the CI tests mention the blanket issue, and at least one fixed it in the PR).

So I guess what I'd like to know is: 

* Is there a plan for publishing this package?
* Are there specific criteria for a new release?
* Are there certain maintainers who are watching/driving this?  

I'd love to see the CI tests (blanket issue) fixed, a quick pass made through the PRs to grab the low-hanging fruit, and a release published.  As someone new to the project, I'm not sure what I can really do to help, but I am happy to help if there is anything I can do.  

FWIW, we needed the OpenStack v3 auth fix, so we tested, then actually deployed that changelist with our product (so in terms of my personal "I think it's ready to ship", I put my money where my mouth is).

**Comment by indexzero** [comment]
@BobDickinson would love your help. To answer your questions:

> It seems like there is still a core group of maintainers who are active. It also seems like the changes are generally maintenance/bug fixes, or adding support for new service functionality within the scope of how and what pkgcloud already provided in terms of API (not a lot of big breaking changes, AFAICT).

Yes. @mmalecki @kenperkins @rosskukulinski and myself are the current maintainers. The API is public uniform API largely stable and imho doesn't need many changes. The underlying API drivers, however, probably need various updates to support new(er) versions that have since been released. 

> So if we have active maintainers and a good set of CI tests, and are essentially in maintenance mode, why no release in 15+ months?

There were several large PRs that were aging on the vine. Since I did not have the adequate time to review them in-depth I merged them into `1.2.0-alpha`. Since that has not seen many bugs up until now I think it is prudent to executing a plan to release it as stable.

> It's also concerning that the CI tests have been failing for 3 months due to a relatively simple issue (the version of blanket, AFAICT), and nobody has dealt with that issue (many recent PRs that fail the CI tests mention the blanket issue, and at least one fixed it in the PR).

We've all largely taken a step back from the project due to other work being more pressing. I've been paying attention to issues as they come in, but everything could use a thorough spring cleaning so to speak.

> Is there a plan for publishing this package?

Yes. Let's get `1.2.0` out there. 

> Are there specific criteria for a new release?

I think that the `alpha.0` will be ready to be stable when two things happen:
1. `CHANGELOG.md` is updated. 
2. Issues opened since `1.2.0-alpha.0` was released (2015/01/07) are triaged and we fix any show stopper bugs that might be present.

Since you mentioned that you've triaged the open issues and you feel strongly we should cut the `1.2.0` release I'm guessing that (2) is largely a non-problem, but I wanted to highlight it.

> Are there certain maintainers who are watching/driving this?

It's largely been just me for the last 12-18 months. I have attempted to summon @mmalecki @kenperkins and @rosskukulinski a few times on other high-profile and/or large PRs, but have not gotten much engagement. As they say "life is what happens when you're busy trying to write Open Source." I've invited you to @pkgcloud/committers. When you're ready for `1.2.0` to be published to `npm` just ping me.

Welcome aboard! :100:

**Comment by BobDickinson** [comment]
Thanks for the detailed response and for adding me to the team.

My original triage was oriented around `master` (the changes I needed got merged there, so I was looking at the suitability of using/releasing `master` when I did my review).  

When I said "please release 1.2.0", I guess what I really meant was "please release master as 1.2.0" (I did not see the 1.2.0 branch or understand the actual 1.2.0 release plan based on that branch).  I will go back and assess the 1.2.0 branch and come up with an opinion about whether anything else needs to be merged there to consider it a stable release.  

That being said, that 1.2.0 release doesn't really solve my problem, which is that I'd like to get the OpenStack v3 auth and the rest of the stuff done in the last year into a current npm release.  So maybe a strategy would be to release 1.2.0 (minimizing changes to that branch), then come up with a 1.3.0 release plan.  I'd probably argue that `master` is suitable for release as 1.3.0 once the CI tests are fixed and handful of clean/easy PRs get merged.

**Comment by indexzero** [comment]
>  (I did not see the 1.2.0 branch or understand the actual 1.2.0 release plan based on that branch).

Yes you are correct: _there is no 1.2.0 branch, it is just master._ There was a 1.2.0 branch a year ago, that's what I was referring to. Sorry for the confusion. 

> That being said, that 1.2.0 release doesn't really solve my problem, which is that I'd like to get the OpenStack v3 auth and the rest of the stuff done in the last year into a current npm release.

Yes. I 100% agree with this strategy. I believe that for all intents and purposes the commit f7799a7d307cd3701eddc3edcf54239806990b09 is `1.2.0`. I will make a tag off of that and publish now. If you could focus on making `HEAD` of `master` an awesome `1.3.0` I think we'll be in good shape very soon.

**Comment by indexzero** [comment]
https://github.com/pkgcloud/pkgcloud/tree/1.2.0 is now published to `npm` as `pkgcloud@1.2.0`.
